nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/diagnostics/bootstrap/tasks/ProcessorTimingDiagnosticTask.java

```diff
@@ -0,0 +1,228 @@
+    //   Read Secs | Write Secs| Commit Sec | GC millis | MB Read    | MB Write   |
+        "%8$16.16s | %9$16.16s | %10$20.20s | %11$13.13s | %12$11.11s | %13$11.11s |";
+
+    public ProcessorTimingDiagnosticTask(final FlowFileEventRepository flowFileEventRepository, final FlowManager flowManager) {
+        this.eventRepo = flowFileEventRepository;
+        this.flowManager = flowManager;
+    }
+
+    @Override
+    public DiagnosticsDumpElement captureDump(final boolean verbose) {
+        final List<String> details = new ArrayList<>();
+
+        final RepositoryStatusReport statusReport = eventRepo.reportTransferEvents(System.currentTimeMillis());
+        final Map<String, FlowFileEvent> eventsByComponentId = statusReport.getReportEntries();
+
+        final List<ProcessorTiming> timings = new ArrayList<>();
+        eventsByComponentId.entrySet().stream()
+            .map(entry -> getTiming(entry.getKey(), entry.getValue()))
+            .filter(Objects::nonNull)
+            .forEach(timings::add); // create ArrayList and add here instead of .collect(toList()) because arraylist allows us to sort
+
+        // Sort based on the Processor CPU time, highest CPU usage first
+        timings.sort(Comparator.comparing(ProcessorTiming::getCpuNanos).reversed());
+
+        final DecimalFormat dataSizeFormat = new DecimalFormat("#,###,###.##");
+        final DecimalFormat percentageFormat = new DecimalFormat("##.##");
+        final NumberFormat secondsFormat = NumberFormat.getInstance();
+
+        long totalCpuNanos = 0L;
+        long totalProcNanos = 0L;
+        long totalReadNanos = 0L;
+        long totalWriteNanos = 0L;
+        long totalSessionCommitNanos = 0L;
+        long totalBytesRead = 0L;
+        long totalBytesWritten = 0L;
+        long totalGcNanos = 0L;
+
+        // Tally totals for all timing elements
+        for (final ProcessorTiming timing : timings) {
+            totalCpuNanos += timing.getCpuNanos();
+            totalProcNanos += timing.getProcessingNanos();
+            totalReadNanos += timing.getReadNanos();
+            totalWriteNanos += timing.getWriteNanos();
+            totalSessionCommitNanos += timing.getSessionCommitNanos();
+            totalBytesRead += timing.getBytesRead();
+            totalBytesWritten += timing.getBytesWritten();
+            totalGcNanos += timing.getGarbageCollectionNanos();
+        }
+
+        if (totalCpuNanos < 1) {
+            details.add("No Processor Timing Diagnostic information has been gathered.");
+            return new StandardDiagnosticsDumpElement("Processor Timing Diagnostics (Stats over last 5 minutes)", details);
+        }
+
+        details.add(String.format(PROCESSOR_TIMING_FORMAT, "Processor ID", "Processor Name", "Processor Type", "Process Group Name", "Processing Secs",
+            "CPU Secs (% time using CPU)", "Pct CPU Time Used by Proc", "Disk Read Secs", "Disk Write Secs", "Session Commit Secs", "GC Millis", "MB Read", "MB Written"));
```

Comment:
Understanding the need to balance readability with length, the current headers and column width produce very long lines in the diagnostic output, making it difficult to read without a very high resolution screen.

What do you think about using shorter names for most of the numeric fields, and then condensing the column width for those fields? Even the `Processor Type` field could be shortened, but the biggest gain would be for the numeric columns.  Perhaps using acronyms that could be listed prior to the tabular output?

```
PS = Processing Seconds
CSP = CPU Seconds and percent time using CPU
PCTP = Percent CPU time used by Processor
DR = Disk Read Seconds
DW = Disk Write Seconds
SC = Session Commit Seconds
GC = Garbage Collection Milliseconds
MR = Megabytes Read
MW = Megabytes Written
```

Reply:
-1. While I can appreciate the desire for conciseness, I think the readability is significantly hindered by reducing the column names to abbreviations. The reality is that if you're using NiFi, you generally are going to have a reasonably large screen, or the canvas is going to be too cramped anyway. And you can always grab the section out and throw it into a spreadsheet if you needed to. Additionally, the width needs to be fairly wide for some of these columns anyway. So having a heading like "MW" for a column that needs to be sufficiently wide to fit something like "4,820,182.54" is a bit awkward as well, when the heading could be explicit.

Reply:
Thanks for the reply @markap14, that's a good point about the potential for larger numbers in column values. In light of the variety of options, and potential for different approaches down the road, leaving it as-is sounds good for now.